osu.Game/Rulesets/Objects/Pooling/PoolableDrawableWithLifetime.cs

```diff
@@ -113,12 +132,12 @@ protected virtual void OnFree(TEntry entry)
 
         private void free()
         {
-            Debug.Assert(Entry != null && HasEntryApplied);
+            Debug.Assert(entry != null && HasEntryApplied);
```

Comment:
Dunno if it's just me but having some of the reads go through `entry` (the field) but some through `Entry` (the property) scares me a bit from a code organisation standpoint, even though I do realise they point at the same reference. I'd probably only permit direct accesses to `entry` (the field) in three cases, namely `Apply()`, `free()`, and the implementation of `Entry` (the property), and have _everything_ else go through `Entry` (the property).

Reply:
One reason I prefer using the same field `entry` consistently within one method is that the code inspect complains about the null possibility when the null checking is done for a property and a backend field is used.
But, in this case, I could use `Entry` here because `entry = null` is okay with possibly null `entry`.
I agree using `Entry` seems nicer for consistency. I made a change.